### PR TITLE
Add AtomicScaleUp method to NodeGroup interface

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind api-change

#### What this PR does / why we need it:
Part of implementation of Atomic (Generic) ProvisioningClass https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md

### CODE REVIEW

**Review by kisieland — APPROVED** [review]
/lgtm

File: cluster-autoscaler/cloudprovider/cloud_provider.go [hunk 1]
```diff
@@ -182,6 +182,10 @@ type NodeGroup interface {
 	// node group size is updated. Implementation required.
 	IncreaseSize(delta int) error
 
+	// AtomicIncreaseSize tries to increase the size of the node group atomically.
+	// If some of nodes couldn't be created, the error is returned. Implementation is optional.
+	AtomicIncreaseSize(delta int) error
```
[the remark this answers is not in the document]
Comment (towca) [reply cluster-autoscaler/cloudprovider/cloud_provider.go:187]: Also similarly to `IncreaseSize`, the method has to ensure that the node group size is updated before returning.

[the remark this answers is not in the document]
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/cloudprovider/cloud_provider.go:194]: Updated

[the remark this answers is not in the document]
Comment (towca) [reply cluster-autoscaler/cloudprovider/cloud_provider.go:194]: Thank you!

**Review by towca — APPROVED** [review]

**Comment by yaroslava-serdiuk** [comment]
#6815

**Pull request merged**